### ISSUE TITLE
FormatListener does not handle empty request correctly

### ISSUE THREAD

**Issue by mpbzh** [FriendsOfSymfony/FOSRestBundle#1215]
If the request accept header is empty, the `FormatListener` calls the `FormatNegotiator` which calls its parent class (`AbstractNegotiator`) like this:
```php
$mimeType = parent::getBest($header, $mimeTypes);
```
(see [here](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/Negotiation/FormatNegotiator.php#L89))

This will cause an `InvalidArgument` exception which is not caught. Here is what the `getBest()` method does:
```php
if (!$header) {
    throw new InvalidArgument('The header string should not be empty.');
}
```
(see [here](https://github.com/willdurand/Negotiation/blob/master/src/Negotiation/AbstractNegotiator.php#L22))

But the only Exception that is caught in the `FormatListener` is the `StopFormatListenerException`.

I think the correct behavior for an empty `Accept` header should be to throw a `NotAcceptableHttpException`, because strictly speaking, the client specifies that he does not accept *any* media type.

Quoting [RFC 2616 Section 14.1](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.1):
> If no Accept header field is present, then it is assumed that the client accepts all media types. If an Accept header field is present, and if the server cannot send a response which is acceptable according to the combined Accept field value, then the server SHOULD send a 406 (not acceptable) response.

**Comment by mpbzh** [comment]
I think the correct behavior for an empty `Accept` header should be to throw a `NotAcceptableHttpException`, because strictly speaking, the client specifies that he does not accept *any* media type.

Quoting [RFC 2616 Section 14.1](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.1):
> If no Accept header field is present, then it is assumed that the client accepts all media types. If an Accept header field is present, and if the server cannot send a response which is acceptable according to the combined Accept field value, then the server SHOULD send a 406 (not acceptable) response.

**Comment by xabbuh** [comment]
fixed in 1af7cab8a315b549b87b605709da486d3384cdd0
